### ISSUE TITLE
是否有计划提供其他数据集的预训练模型，用作通用分割？是否会支持高版本torch？

### ISSUE THREAD

**Issue by yski** [spiderforest/MixerCSeg#3]
首先，感谢作者无私奉献！
本项目在crack数据集上确实很能打。按理说在其他数据集应该也表现不错，作者是否有计划测试其他数据集？
torchaudio==2.1.0+cu118是不支持50系列GPU的，我测试用2.8+CUDA12.9无法启用selective_scan，triton也失败了,好像会影响速度。
 D:\XYAI_20250325\MixerCSeg [1:1] $ python main.py --dataset_path ./DeepCrack --batch_size_train 1 --batch_size_test 1 --epochs 50 --lr 5e-4 --device cuda --dataset_mode crack --phase train
D:\XYAI_20250325\SegMAN\mmcv\mmcv\__init__.py:20: UserWarning: On January 1, 2023, MMCV will release v2.0.0, in which it will remove components related to the training process and add a data transformation module. In addition, it will rename the package names mmcv to mmcv-lite and mmcv-full to mmcv. See https://github.com/open-mmlab/mmcv/blob/master/docs/en/compatibility.md for more details.
  warnings.warn(
D:\XYAI_20250325\MixerCSeg\VMamba\models\csm_triton.py:11: UserWarning: Triton not installed, fall back to pytorch implements.
  warnings.warn("Triton not installed, fall back to pytorch implements.")
D:\XYAI_20250325\MixerCSeg\VMamba\models\csms6s.py:14: UserWarning: Can not import selective_scan_cuda_oflex. This affects speed.
  warnings.warn("Can not import selective_scan_cuda_oflex. This affects speed.")
Can not import selective_scan_cuda_oflex. This affects speed.
D:\XYAI_20250325\MixerCSeg\VMamba\models\csms6s.py:75: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.   
  @torch.cuda.amp.custom_fwd
D:\XYAI_20250325\MixerCSeg\VMamba\models\csms6s.py:99: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.   
  @torch.cuda.amp.custom_bwd
args: BCELoss_ratio -> 0.87
args: DiceLoss_ratio -> 0.13
The number of training images = 368
use AdamW!
Start processing!
---------------------------------------------------------------------------------------
training epoch start ->  0
Initial Loss Fused: Pending:   0%|                                       | 0/368 [00:00<?, ?it/s]D:\XYAI_20250325\SegMAN\mmcv\mmcv\__init__.py:20: UserWarning: On January 1, 2023, MMCV will release v2.0.0, in which it will remove components related to the training process and add a data transformation module. In addition, it will rename the package names mmcv to mmcv-lite and mmcv-full to mmcv. See https://github.com/open-mmlab/mmcv/blob/master/docs/en/compatibility.md for more details.
  warnings.warn(
D:\XYAI_20250325\MixerCSeg\VMamba\models\csm_triton.py:11: UserWarning: Triton not installed, fall back to pytorch implements.
  warnings.warn("Triton not installed, fall back to pytorch implements.")
D:\XYAI_20250325\MixerCSeg\VMamba\models\csms6s.py:14: UserWarning: Can not import selective_scan_cuda_oflex. This affects speed.
  warnings.warn("Can not import selective_scan_cuda_oflex. This affects speed.")
Can not import selective_scan_cuda_oflex. This affects speed.
D:\XYAI_20250325\MixerCSeg\VMamba\models\csms6s.py:75: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.   
  @torch.cuda.amp.custom_fwd
D:\XYAI_20250325\MixerCSeg\VMamba\models\csms6s.py:99: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.   
  @torch.cuda.amp.custom_bwd
Loss: 0.5410:  10%|████▎                                      | 37/368 [29:16<4:25:02, 48.04s/it]

**Comment by spiderforest** [comment]
感谢你对我们项目的关注。
1. 我确实有想过将TranMixer架构进行进一步的优化并扩展到更广泛的场景，但我目前已经离开了原来的实验室，加入了新的实验室，我暂时还无法确定是否有足够的时间来完成这件事。
2. 我没有调试过50系列的GPU，可能无法帮助到你。
3. selective_scan/triton 适配的问题你可以在[VMamba](https://github.com/MzeroMiko/VMamba)、[Mamba](https://github.com/state-spaces/mamba)寻找一下是否有相关解决方案；或者尝试在50系列显卡安装低版本的CUDA。希望这些能帮助到你
